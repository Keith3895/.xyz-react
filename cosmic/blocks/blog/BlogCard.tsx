import { getFormattedDate } from "../../../cosmic/utils";
import BlogCardComponent from "../../../components/blogCard";

export type PostType = {
  id: string
  title: string
  slug: string
  metadata: {
    categories: any
    image: {
      imgix_url: string
    }
    content: string
    author: {
      title: string
      metadata: {
        seo:{
          description: string
        }
        image: {
          imgix_url: string
        }
      }
    }
    published_date: string
  }
}

export function BlogCard({
  post,
  className,
}: {
  post: PostType
  className?: string
}) {
  return (
    <BlogCardComponent title={post.title} thumbnail={post.metadata.image.imgix_url} category={post.metadata.categories[0].title} date={getFormattedDate(post.metadata.published_date)} slug={post.slug} description={post.metadata.seo.description} />
  )
}
