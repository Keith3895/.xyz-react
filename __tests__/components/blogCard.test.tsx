import { render, screen } from '@testing-library/react';
import BlogCard from '../../src/components/blogCard';

describe('BlogCard', () => {
    const blog = {
        title: 'Test Title',
        description: 'Test Description',
        thumbnail: 'https://imgix.cosmicjs.com/cb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg',
        category: 'Test Category',
        date: 'Test Date',
        slug: 'test-slug',
    };

    it('renders the blog thumbnail', () => {
        render(<BlogCard {...blog} />);
        const img = screen.getByRole('img');
        // expect(img).toHaveAttribute('src', blog.thumbnail);
        expect(img).toHaveAttribute('src', `/_next/image?url=${encodeURIComponent(blog.thumbnail)}&w=1080&q=100`);
    });

    it('renders the blog category', () => {
        render(<BlogCard {...blog} />);
        expect(screen.getByText(blog.category)).toBeInTheDocument();
    });

    it('renders the blog date', () => {
        render(<BlogCard {...blog} />);
        expect(screen.getByText(blog.date)).toBeInTheDocument();
    });

    it('renders the blog title', () => {
        render(<BlogCard {...blog} />);
        expect(screen.getByText(blog.title)).toBeInTheDocument();
    });

    it('renders the blog description', () => {
        render(<BlogCard {...blog} />);
        expect(screen.getByText(blog.description)).toBeInTheDocument();
    });

    it('renders the "Read more" link with correct href', () => {
        render(<BlogCard {...blog} />);
        const link = screen.getByRole('link', { name: /read more/i });
        expect(link).toHaveAttribute('href', `blog/${blog.slug}`);
    });

    it('matches the snapshot', () => {
        const { container } = render(<BlogCard {...blog} />);
        expect(container).toMatchSnapshot();
    });

    it('does not render the image if thumbnail is not provided', () => {
        const blogNeg = {
            title: 'Test Title',
            description: 'Test Description',
            category: 'Test Category',
            date: 'Test Date',
            slug: 'test-slug',
        };
        render(<BlogCard {...blogNeg} />);
        expect(screen.queryByRole('img')).toBeInTheDocument();
        const img = screen.queryByRole('img') as HTMLImageElement;
        expect(img.src).toBe('http://localhost/');
    });
    it('renders homepage unchanged', () => {
        const { container } = render(<BlogCard {...blog} />);
        expect(container).toMatchSnapshot()
      });
});