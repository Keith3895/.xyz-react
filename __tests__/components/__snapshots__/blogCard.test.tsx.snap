// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard matches the snapshot 1`] = `
<div>
  <div
    class="md:max-w-sm w-11/12 bg-white border border-gray-200 rounded-lg shadow md:w-3/12"
  >
    <a
      class="justify-center flex flex-row item-center"
      href="blog/test-slug"
    >
      <img
        alt=""
        class="rounded-t-lg"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimgix.cosmicjs.com%2Fcb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg&w=1080&q=100"
        srcset="/_next/image?url=https%3A%2F%2Fimgix.cosmicjs.com%2Fcb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg&w=640&q=100 1x, /_next/image?url=https%3A%2F%2Fimgix.cosmicjs.com%2Fcb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg&w=1080&q=100 2x"
        style="color: transparent;"
        width="500"
      />
    </a>
    <div
      class="p-5 flex flex-row space-x-4"
    >
      <div
        class="text-xs font-normal "
      >
         
        Test Category
         
      </div>
      <div
        class="text-center text-zinc-500 text-xs font-normal "
      >
        Test Date
      </div>
    </div>
    <div
      class="p-5"
    >
      <a
        href="blog/test-slug"
      >
        <h5
          class="mb-2 text-2xl font-bold tracking-tight text-gray-900"
        >
          Test Title
        </h5>
      </a>
      <p
        class="mb-3 font-normal text-gray-700 line-clamp-3"
      >
        Test Description
      </p>
      <a
        class="inline-flex items-center px-3 py-2 text-sm font-medium text-center  text-black bg-custom-orange-700 rounded-lg hover:bg-custom-orange-800 focus:ring-4  focus:outline-none focus:ring-custom-orange-300"
        href="blog/test-slug"
      >
        Read more
        <svg
          aria-hidden="true"
          class="rtl:rotate-180 w-3.5 h-3.5 ms-2"
          fill="none"
          viewBox="0 0 14 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 5h12m0 0L9 1m4 4L9 9"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`BlogCard renders homepage unchanged 1`] = `
<div>
  <div
    class="md:max-w-sm w-11/12 bg-white border border-gray-200 rounded-lg shadow md:w-3/12"
  >
    <a
      class="justify-center flex flex-row item-center"
      href="blog/test-slug"
    >
      <img
        alt=""
        class="rounded-t-lg"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimgix.cosmicjs.com%2Fcb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg&w=1080&q=100"
        srcset="/_next/image?url=https%3A%2F%2Fimgix.cosmicjs.com%2Fcb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg&w=640&q=100 1x, /_next/image?url=https%3A%2F%2Fimgix.cosmicjs.com%2Fcb1152e0-ccd3-11ee-a584-65599426e889-Our-Amazing-Adventure.jpg&w=1080&q=100 2x"
        style="color: transparent;"
        width="500"
      />
    </a>
    <div
      class="p-5 flex flex-row space-x-4"
    >
      <div
        class="text-xs font-normal "
      >
         
        Test Category
         
      </div>
      <div
        class="text-center text-zinc-500 text-xs font-normal "
      >
        Test Date
      </div>
    </div>
    <div
      class="p-5"
    >
      <a
        href="blog/test-slug"
      >
        <h5
          class="mb-2 text-2xl font-bold tracking-tight text-gray-900"
        >
          Test Title
        </h5>
      </a>
      <p
        class="mb-3 font-normal text-gray-700 line-clamp-3"
      >
        Test Description
      </p>
      <a
        class="inline-flex items-center px-3 py-2 text-sm font-medium text-center  text-black bg-custom-orange-700 rounded-lg hover:bg-custom-orange-800 focus:ring-4  focus:outline-none focus:ring-custom-orange-300"
        href="blog/test-slug"
      >
        Read more
        <svg
          aria-hidden="true"
          class="rtl:rotate-180 w-3.5 h-3.5 ms-2"
          fill="none"
          viewBox="0 0 14 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 5h12m0 0L9 1m4 4L9 9"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
